modules/account-lib/src/coin/sol/keyPair.ts

```diff
@@ -1,34 +1,120 @@
-import { Ed25519KeyPair } from '../baseCoin';
-import { NotImplementedError } from '../baseCoin/errors';
-import { DefaultKeys, KeyPairOptions } from '../baseCoin/iface';
+import { Keypair as SolKeypair, PublicKey } from '@solana/web3.js';
+import nacl from 'tweetnacl';
+import { BaseKeyPair } from '../baseCoin';
+import { InvalidKey } from '../baseCoin/errors';
+import { DefaultKeys, isPrivateKey, isPublicKey, isSeed, KeyPairOptions } from '../baseCoin/iface';
+import { SEED_LENGTH } from './constants';
+import { SolanaKeys } from './iface';
+import { base58ToUint8Array, isValidPrivateKey, isValidPublicKey, Uint8ArrayTobase58 } from './utils';
 
-export class KeyPair extends Ed25519KeyPair {
+export class KeyPair implements BaseKeyPair {
```

Comment:
why did we need to change this? Solana key pairs are ed25519.

Reply:
Mostly b/c the Ed25519KeyPair implementation converts keys to hex instead of base58 and i liked to use the Solana lib instead of nacl for address generation.

Reply:
add comment explaining this is still a ed25519 key pair for documentation.
also add a TODO to refactor `Ed25519KeyPair` to support encoding changes (e.g. hex or base58)